### DISCUSSION TITLE
CMake VerifyFortran name mangling issue (CMake-2.28.3, GCC is 12.2.1 provided by my Oracle Linux 8 system.)

### DISCUSSION THREAD

**Discussion by bgusto** [Nek5000/nekRS#548] [Compiling] [answered]
Hi, I'm just throwing this out there to see if anyone's run into this before. CMake tests throw an error compiling the test Fortran programs because of a name mangling issue. My OpenMPI is version 5.0.2 (latest stable) and was built with GCC-12.2.1 by me. Happy to provide more details if anyone is interested.

[bgusto@galadriel nekRS]$ CC=mpicc CXX=mpic++ FC=mpif77 ./nrsconfig -DCMAKE_INSTALL_PREFIX=$NEKRS_HOME/build
cmake -S . -B build -Wfatal-errors -DCMAKE_INSTALL_PREFIX=/home/bgusto/Software/nekRS/build
-- The C compiler identification is GNU 12.2.1
-- The CXX compiler identification is GNU 12.2.1
-- The Fortran compiler identification is GNU 8.5.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpicc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpic++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpif77 - skipped
-- Detecting Fortran/C Interface
Failed to compile
-- Found MPI_C: /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpicc (found version "3.1") 
-- Found MPI_CXX: /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpic++ (found version "3.1") 
-- Found MPI_Fortran: /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpif77 (found version "3.1") 
-- Found MPI: TRUE (found version "3.1")  
-- Verifying Fortran/CXX Compiler Compatibility
CMake Warning (dev) at /opt/cmake-3.28.3/Modules/FortranCInterface.cmake:309 (message):
  No FortranCInterface mangling known for VerifyFortran
Call Stack (most recent call first):
  /opt/cmake-3.28.3/Modules/FortranCInterface/Verify/CMakeLists.txt:16 (FortranCInterface_HEADER)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Verifying Fortran/CXX Compiler Compatibility - Failed
CMake Warning at CMakeLists.txt:59 (message):
  C and Fortran may not work properly

-- Found MPI_UNDERLYING_COMPILER: /opt/rh/gcc-toolset-12/root/usr/bin/g++
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP_Fortran: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- Performing Test Allowed_LD_Flag_BSYMBOLIC
-- Performing Test Allowed_LD_Flag_BSYMBOLIC - Failed
-- Using CMake version 3.28.3
-- Performing Test Allowed_CXX_Flag_Wall__Wextra
-- Performing Test Allowed_CXX_Flag_Wall__Wextra - Success
-- Performing Test Allowed_CXX_Flag_Wunused_function__Wunused_variable
-- Performing Test Allowed_CXX_Flag_Wunused_function__Wunused_variable - Success
-- Performing Test Allowed_CXX_Flag_Wwrite_strings__Wfloat_equal
-- Performing Test Allowed_CXX_Flag_Wwrite_strings__Wfloat_equal - Success
-- Performing Test Allowed_CXX_Flag_Wcast_align__Wlogical_op
-- Performing Test Allowed_CXX_Flag_Wcast_align__Wlogical_op - Success
-- Performing Test Allowed_CXX_Flag_Wshadow
-- Performing Test Allowed_CXX_Flag_Wshadow - Success
-- Performing Test Allowed_CXX_Flag_Wno_unused_parameter
-- Performing Test Allowed_CXX_Flag_Wno_unused_parameter - Success
-- Performing Test Allowed_CXX_Flag_diag_disable_11074__diag_disable_11076
-- Performing Test Allowed_CXX_Flag_diag_disable_11074__diag_disable_11076 - Failed
-- Performing Test Allowed_CXX_Flag__display_error_number
-- Performing Test Allowed_CXX_Flag__display_error_number - Failed
-- Performing Test Allowed_CXX_Flag_Werror
-- Performing Test Allowed_CXX_Flag_Werror - Success
-- Performing Test Allowed_C_Flag_Wall__Wextra
-- Performing Test Allowed_C_Flag_Wall__Wextra - Success
-- Performing Test Allowed_C_Flag_Wunused_function__Wunused_variable
-- Performing Test Allowed_C_Flag_Wunused_function__Wunused_variable - Success
-- Performing Test Allowed_C_Flag_Wwrite_strings__Wfloat_equal
-- Performing Test Allowed_C_Flag_Wwrite_strings__Wfloat_equal - Success
-- Performing Test Allowed_C_Flag_Wcast_align__Wlogical_op
-- Performing Test Allowed_C_Flag_Wcast_align__Wlogical_op - Success
-- Performing Test Allowed_C_Flag_Wshadow
-- Performing Test Allowed_C_Flag_Wshadow - Success
-- Performing Test Allowed_C_Flag_Wno_c__11_long_long
-- Performing Test Allowed_C_Flag_Wno_c__11_long_long - Success
-- Performing Test Allowed_C_Flag_diag_disable_11074__diag_disable_11076
-- Performing Test Allowed_C_Flag_diag_disable_11074__diag_disable_11076 - Failed
-- Performing Test Allowed_C_Flag__display_error_number
-- Performing Test Allowed_C_Flag__display_error_number - Failed
-- Performing Test Allowed_C_Flag_Werror
-- Performing Test Allowed_C_Flag_Werror - Success
-- Performing Test COMPILER_SUPPORTS_NO_STRICT_ALIASING
-- Performing Test COMPILER_SUPPORTS_NO_STRICT_ALIASING - Success
-- System     : Linux-4.18.0-477.15.1.el8_8.x86_64
-- Build type : RelWithDebInfo
-- C flags    :  -Wall -Wextra -Wunused-function -Wunused-variable -Wwrite-strings -Wfloat-equal -Wcast-align -Wlogical-op -Wshadow -Wno-c++11-long-long  -fopenmp -O2 -g -march=native -mtune=native -ftree-vectorize
-- CXX flags  :  -Wall -Wextra -Wunused-function -Wunused-variable -Wwrite-strings -Wfloat-equal -Wcast-align -Wlogical-op -Wshadow -Wno-unused-parameter  -fopenmp -fno-strict-aliasing -O2 -g -march=native -mtune=native -ftree-vectorize
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP_Fortran: -fopenmp (found version "4.5") 
-- OpenMP include dirs: 
-- OpenMP libraries:    /opt/rh/gcc-toolset-12/root/usr/lib/gcc/x86_64-redhat-linux/12/libgomp.so
-- Could not find nvcc, please set CUDAToolkit_ROOT.
-- Looking for DPC++
-- Could NOT find DPCPP (missing: SYCL_INCLUDE_DIRS SYCL_LIBRARIES) 
-- Could NOT find HIP (missing: HIP_ROOT_DIR HIP_INCLUDE_DIRS HIP_LIBRARIES HIP_RUNTIME_DEFINE HIP_HIPCONFIG_EXECUTABLE HIP_PLATFORM HIP_COMPILER) 
-- OCCA_CUDA_ENABLED: 0
-- OCCA_HIP_ENABLED: 0
-- OCCA_DPCPP_ENABLED: 0
-- OCCA_OPENCL_ENABLED: 
CMake Deprecation Warning at 3rd_party/lapack/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- Checking if build type is 'Coverage'
-- Performing Test _frecursiveFlag
-- Performing Test _frecursiveFlag - Success
-- Build tests: OFF
-- Reducing RELEASE optimization level to O2
CMake Deprecation Warning at /home/bgusto/Software/nekRS/3rd_party/lapack/INSTALL/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- Looking for Fortran NONE - found
CMake Deprecation Warning at /home/bgusto/Software/nekRS/3rd_party/lapack/INSTALL/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- Looking for Fortran INT_CPU_TIME - found
CMake Deprecation Warning at /home/bgusto/Software/nekRS/3rd_party/lapack/INSTALL/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- Looking for Fortran EXT_ETIME - not found
CMake Deprecation Warning at /home/bgusto/Software/nekRS/3rd_party/lapack/INSTALL/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- Looking for Fortran EXT_ETIME_ - not found
CMake Deprecation Warning at /home/bgusto/Software/nekRS/3rd_party/lapack/INSTALL/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.

-- Looking for Fortran INT_ETIME - found
-- --> Will use second_INT_ETIME.f and dsecnd_INT_ETIME.f as timing function.
-- Build deprecated routines: OFF
-- Build single precision real: OFF
-- Build double precision real: ON
-- Build single precision complex: OFF
-- Build double precision complex: OFF
-- Using supplied NETLIB BLAS implementation
-- Using supplied NETLIB LAPACK implementation
-- Verifying Fortran/C Compiler Compatibility
CMake Warning (dev) at /opt/cmake-3.28.3/Modules/FortranCInterface.cmake:309 (message):
  No FortranCInterface mangling known for VerifyFortran
Call Stack (most recent call first):
  /opt/cmake-3.28.3/Modules/FortranCInterface/Verify/CMakeLists.txt:16 (FortranCInterface_HEADER)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Verifying Fortran/C Compiler Compatibility - Failed
CMake Error at /opt/cmake-3.28.3/Modules/FortranCInterface.cmake:398 (message):
  The Fortran compiler:

    /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpif77

  and the C compiler:

    /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpicc

  failed to compile a simple test project using both languages.  The output
  was:

    Change Dir: '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    
    Run Build Command(s): /opt/cmake-3.28.3/bin/cmake -E env VERBOSE=1 /usr/bin/gmake -f Makefile VerifyFortranC
    /opt/cmake-3.28.3/bin/cmake -S/opt/cmake-3.28.3/Modules/FortranCInterface/Verify -B/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC --check-build-system CMakeFiles/Makefile.cmake 0
    /usr/bin/gmake  -f CMakeFiles/Makefile2 VerifyFortranC
    gmake[1]: Entering directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    /opt/cmake-3.28.3/bin/cmake -S/opt/cmake-3.28.3/Modules/FortranCInterface/Verify -B/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC --check-build-system CMakeFiles/Makefile.cmake 0
    /opt/cmake-3.28.3/bin/cmake -E cmake_progress_start /home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles 5
    /usr/bin/gmake  -f CMakeFiles/Makefile2 CMakeFiles/VerifyFortranC.dir/all
    gmake[2]: Entering directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    /usr/bin/gmake  -f CMakeFiles/VerifyFortran.dir/build.make CMakeFiles/VerifyFortran.dir/depend
    gmake[3]: Entering directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    cd /home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC && /opt/cmake-3.28.3/bin/cmake -E cmake_depends "Unix Makefiles" /opt/cmake-3.28.3/Modules/FortranCInterface/Verify /opt/cmake-3.28.3/Modules/FortranCInterface/Verify /home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC /home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC /home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/VerifyFortran.dir/DependInfo.cmake
    Dependee "/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/VerifyFortran.dir/DependInfo.cmake" is newer than depender "/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/VerifyFortran.dir/depend.internal".
    Dependee "/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/VerifyFortran.dir/depend.internal".
    Scanning dependencies of target VerifyFortran
    gmake[3]: Leaving directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    /usr/bin/gmake  -f CMakeFiles/VerifyFortran.dir/build.make CMakeFiles/VerifyFortran.dir/build
    gmake[3]: Entering directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    [ 20%] Building Fortran object CMakeFiles/VerifyFortran.dir/VerifyFortran.f.o
    /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpif77  -I/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC -frecursive -O2 -DNDEBUG -O2 -c /opt/cmake-3.28.3/Modules/FortranCInterface/Verify/VerifyFortran.f -o CMakeFiles/VerifyFortran.dir/VerifyFortran.f.o
    [ 40%] Linking Fortran static library libVerifyFortran.a
    /opt/cmake-3.28.3/bin/cmake -P CMakeFiles/VerifyFortran.dir/cmake_clean_target.cmake
    /opt/cmake-3.28.3/bin/cmake -E cmake_link_script CMakeFiles/VerifyFortran.dir/link.txt --verbose=1
    /opt/rh/gcc-toolset-12/root/usr/bin/ar qc libVerifyFortran.a CMakeFiles/VerifyFortran.dir/VerifyFortran.f.o
    /opt/rh/gcc-toolset-12/root/usr/bin/ranlib libVerifyFortran.a
    gmake[3]: Leaving directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    [ 40%] Built target VerifyFortran
    /usr/bin/gmake  -f CMakeFiles/VerifyFortranC.dir/build.make CMakeFiles/VerifyFortranC.dir/depend
    gmake[3]: Entering directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    cd /home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC && /opt/cmake-3.28.3/bin/cmake -E cmake_depends "Unix Makefiles" /opt/cmake-3.28.3/Modules/FortranCInterface/Verify /opt/cmake-3.28.3/Modules/FortranCInterface/Verify /home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC /home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC /home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/VerifyFortranC.dir/DependInfo.cmake
    Dependee "/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/VerifyFortranC.dir/DependInfo.cmake" is newer than depender "/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/VerifyFortranC.dir/depend.internal".
    Dependee "/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC/CMakeFiles/VerifyFortranC.dir/depend.internal".
    Scanning dependencies of target VerifyFortranC
    gmake[3]: Leaving directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    /usr/bin/gmake  -f CMakeFiles/VerifyFortranC.dir/build.make CMakeFiles/VerifyFortranC.dir/build
    gmake[3]: Entering directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    [ 60%] Building C object CMakeFiles/VerifyFortranC.dir/main.c.o
    /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpicc  -I/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC -fopenmp -O3 -DNDEBUG -o CMakeFiles/VerifyFortranC.dir/main.c.o -c /opt/cmake-3.28.3/Modules/FortranCInterface/Verify/main.c
    [ 80%] Building C object CMakeFiles/VerifyFortranC.dir/VerifyC.c.o
    /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpicc  -I/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC -fopenmp -O3 -DNDEBUG -o CMakeFiles/VerifyFortranC.dir/VerifyC.c.o -c /opt/cmake-3.28.3/Modules/FortranCInterface/Verify/VerifyC.c
    [100%] Linking C executable VerifyFortranC
    /opt/cmake-3.28.3/bin/cmake -E cmake_link_script CMakeFiles/VerifyFortranC.dir/link.txt --verbose=1
    /opt/openmpi-5.0.2-gcc-12.2.1/bin/mpicc  -fopenmp -O3 -DNDEBUG CMakeFiles/VerifyFortranC.dir/main.c.o CMakeFiles/VerifyFortranC.dir/VerifyC.c.o -o VerifyFortranC   -L/usr/lib/gcc/x86_64-redhat-linux/8  libVerifyFortran.a -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lgfortran -lm -lquadmath -lm 
    /opt/rh/gcc-toolset-12/root/usr/libexec/gcc/x86_64-redhat-linux/12/ld: CMakeFiles/VerifyFortranC.dir/main.c.o: in function `main':
    main.c:(.text.startup+0xa): undefined reference to `VerifyFortran'
    collect2: error: ld returned 1 exit status
    gmake[3]: *** [CMakeFiles/VerifyFortranC.dir/build.make:115: VerifyFortranC] Error 1
    gmake[3]: Leaving directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    gmake[2]: *** [CMakeFiles/Makefile2:114: CMakeFiles/VerifyFortranC.dir/all] Error 2
    gmake[2]: Leaving directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    gmake[1]: *** [CMakeFiles/Makefile2:121: CMakeFiles/VerifyFortranC.dir/rule] Error 2
    gmake[1]: Leaving directory '/home/bgusto/Software/nekRS/build/CMakeFiles/FortranCInterface/VerifyC'
    gmake: *** [Makefile:140: VerifyFortranC] Error 2
    
    
Call Stack (most recent call first):
  3rd_party/lapack/LAPACKE/CMakeLists.txt:4 (FortranCInterface_VERIFY)

**Comment by stgeke** [comment, marked as answer]
```
-- The C compiler identification is GNU 12.2.1
-- The CXX compiler identification is GNU 12.2.1
-- The Fortran compiler identification is GNU 8.5.0
```
Is there any reason to use a different version for gfortran?

**Reply by bgusto** [reply]
No, I missed that, thank you. I had not noticed gfortran was not included in the gcc toolset I downloaded. I installed the toolset with gfortran using 'dnf install gcc-toolset-12-gcc-gfortran-12.2.1-7.4.0.2.el8.x86_64.' Did not get a libocca.so file out of the cmake process. Is CUDA a requirement? I'll add my screen output in a bit in case you might wish to take a look. Thanks.

**Reply by stgeke** [reply]
What happens if you run:
```
cd build
make libocca 
ls lib 
```

**Reply by bgusto** [reply]
I have the libocca.so file there now, however when running example problem in the README I get

MPI tasks: 2

reading par file ...
Initializing device 
active occa mode: Serial


---[ Error ]--------------------------------------------------------------------
    File     : /home/bgusto/Software/nekRS/3rd_party/occa/src/occa/internal/io/utils.cpp
    Line     : 375
    Function : c_read
    Message  : Failed to open [/home/bgusto/Software/nekRS/build/include/occa/scripts/findCompilerVendor.cpp]
    Stack
      13 occa::error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
      12 occa::io::c_read(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long*, occa::enums::FileType)
      11 occa::hashFile(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
      10 occa::sys::compilerVendor(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
       9 occa::serial::device::buildKernel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::hash_t, occa::json const&, bool)
       8 occa::serial::device::buildKernel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::hash_t, occa::json const&)
       7 occa::device::buildKernel(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::json const&) const
       6 occa::device::buildKernelFromString(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, occa::json const&) const
       5 platform_t::platform_t(setupAide&, ompi_communicator_t*, ompi_communicator_t*)
       4 nekrs::setup(ompi_communicator_t*, ompi_communicator_t*, int, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, int, int)
       3 nekrs(main+0xe0a)
       2 /lib64/libc.so.6(__libc_start_main+0xe5)
       1 nekrs(_start+0x2e)
================================================================================

**Reply by stgeke** [reply]
How did you "fix" the occa build issue you ran into before?
